City-of-Helsinki/helfi-kymp-district-project-search
# UHF-6216: search filters

# [UHF-6216 Filters](https://helsinkisolutionoffice.atlassian.net/browse/UHF-6216)

## What was done?

* Add search filters
* Update packages
* Add and enhance components, types, hooks, helpers

## How to install
* Check install instructions for Drupal here: https://github.com/City-of-Helsinki/drupal-helfi-kymp/pull/477
* ... OR: `nvm use; npm i; npm start` for the standalone app

## How to test
* Check that app looks like sketch (https://share.goabstract.com/833a97f8-a5b7-423b-9e3b-461fe6c0dcc6?collectionLayerId=cedd7e0d-a388-418c-9825-4b1c18cc7844&mode=build)
* Search functionality things to test for:
  * Filters components should work. Check filter rules from https://helsinkisolutionoffice.atlassian.net/browse/UHF-6216
  * Filters and results should consider the language context
  * Query should be executed only when pressing the submit button
  * No results should render 'no results' text
  * Changing the filter values should be represented in the url query string
  * Pagination should work

## Designer check

- [ ] This branch has been checked by the designer

## Other PRs


https://github.com/City-of-Helsinki/drupal-helfi-kymp/pull/477
